mochify/client.js

```diff
@@ -133,17 +133,25 @@
   };
   mocha.mochify_run = function () {
     // Inject script
-    var s = document.createElement('script');
-    s.type = 'text/javascript';
-    s.textContent = chunks.join('');
-    document.body.appendChild(s);
+    if (chunks.length) {
+      var s = document.createElement('script');
+      s.type = 'text/javascript';
+      s.textContent = chunks.join('');
+      document.body.appendChild(s);
+    }
     // Run mocha
-    mocha.run(function (code) {
-      if (typeof __coverage__ !== 'undefined') {
-        write('mochify.coverage', window.__coverage__);
-      }
-      write('mochify.callback', { code: code });
-    });
+    Promise.all(window.mochify_pending || [])
```

Comment:
Do all browsers we need to support (you downgraded the client file recently) support native Promises?

Reply:
I guess they don't. I wanted to get things running down to IE, since SinonJS still has that supported.

We might say we don't want to go that far back and allow ourselves to feel like it's 2021 😁

Anyway, I think there's a misunderstanding in where this is implemented. I thought the script tags would be generated in start page of the http server. Do you think it's better to do this in the client?

Reply:
Yeah I initially understood it just like you described too, but then I found it meant we'd have to reimplement it over and over at driver level (in case I do understand it correctly that is 🙉 ). Like this it could happen at Mochify level once.

Then again, doing it at driver level of course would also solve the problem of knowing if the driver can actually support it.

---

That being said, I think we'd need that promise like thing even in case we implemented this feature at driver level as we'd still need to make sure script tags have loaded all of their dependencies before we run Mocha. Or is that incorrect?

Reply:
Or would using `esm` mean we generate a templated index document in the API module that is served by the server and then used as the host for running the test cases?

Reply:
Yeah, I though we generate the script tags in the server in `@mochify/mochify`. Driver independent and without changing the client.

Reply:
It sounds good although I am not entirely sure we can get away without having to change the client. Let me try doing that.

Reply:
Turns out you were right (I don't understand why module loading happens asynchronously in case you append a script tag though) and we don't need to touch the client when we do it at server level. Feel free to have another look and let me know what you think.